SRP/HourReporter.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+require_once("./EmployeeData.php");
+
+/**
+ * 人事クラス
+ */
+class HourReporter extends Employee
+{
+    /**
+     * 期間内の労働時間をレポートします。
+     *
+     * @return integer 月間労働時間
+     */
+    private function reportHours(): string
+    {
+        // 総合で所定労働時間を超えた時間が残業時間
+        $overWorkHour = $this->workHour - ($this->regularHoursHourReporter() * $this->workDayCount);
+
+        $text = "社員番号: " . $this->id . "\n";
+        $text .= "お名前: " . $this->name . " さん\n";
+        $text .= "\n";
+        $text  .= "総業務時間: " . $this->workHour . " 時間 \n";
+        $text .= "残業時間: " . ($overWorkHour > 0 ? $overWorkHour : 0) . "時間 \n";
+        if ($overWorkHour > 10) {
+            $text .= "\n！！残業時間が 10 時間を超えています！！\n";
+        }
+        return $text;
+    }
+    /**
+     * 所定労働時間算出
+     *
+     * @return integer 所定労働時間
+     */
+    protected function regularHoursHourReporter(): float
```

Comment:
```suggestion
    private function regularHours(): float
```
2 点あります。

* このクラス内の利用で閉じたいので、アクセス修飾子は private にしていただきたいです。
* このクラス内の利用で閉じたいので、HourReporter という接尾語は不要です。